website/docs/r/cdn_frontdoor_route.html.markdown

```diff
@@ -0,0 +1,179 @@
+  }
+}
+
+resource "azurerm_cdn_frontdoor_route" "example" {
+  name                            = "example-route"
+  cdn_frontdoor_endpoint_id       = azurerm_cdn_frontdoor_endpoint.example.id
+  cdn_frontdoor_origin_group_id   = azurerm_cdn_frontdoor_origin_group.example.id
+  cdn_frontdoor_custom_domain_ids = [azurerm_cdn_frontdoor_custom_domain.contoso.id, azurerm_cdn_frontdoor_custom_domain.fabrikam.id]
+  cdn_frontdoor_origin_ids        = [azurerm_cdn_frontdoor_origin.example.id]
+  cdn_frontdoor_rule_set_ids      = [azurerm_cdn_frontdoor_rule_set.example.id]
+  enabled                         = true
+
+  forwarding_protocol            = "HttpsOnly"
+  https_redirect_enabled         = true
+  link_to_default_domain_enabled = false
+  patterns_to_match              = ["/*"]
+  supported_protocols            = ["Http", "Https"]
+
+  cache {
+    query_string_caching_behavior = "IgnoreSpecifiedQueryStrings"
+    query_strings                 = ["account", "settings"]
+    compression_enabled           = true
+    content_types_to_compress     = ["text/html", "text/javascript", "text/xml"]
+  }
+}
+```
+
+## Arguments Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The name which should be used for this Frontdoor Route. Valid values must begin with a letter or number, end with a letter or number and may only contain letters, numbers and hypens with a maximum length of 90 characters. Changing this forces a new Frontdoor Route to be created.
+
+* `cdn_frontdoor_endpoint_id` - (Required) The ID of the Frontdoor Route. Changing this forces a new Frontdoor Route to be created.
+
+* `cdn_frontdoor_origin_group_id` - (Required) The resource ID of the Frontdoor Origin Group.
+
+* `cdn_frontdoor_origin_ids` - (Required) One or more Frontdoor Origin resource IDs that this Frontdoor Route will link to.
+
+* `supported_protocols` - (Required) One or more Protocols supported by this Frontdoor Route. Possible values are `Http` or `Https`.
+
+* `patterns_to_match` - (Reqired) The route patterns of the rule.
+
+* `cdn_frontdoor_custom_domain_ids` - (Optional) One or more resource IDs of the Frontdoor Custom Domains to associate with the Frontdoor Route.
+
+* `link_to_default_domain_enabled` - (Optional) Will the Frontdoor Route be linked to the default domain endpoint? Possible values are `true` or `false`. Defaults to `false`.
+
+* `cache` - (Optional) A `cache` block as defined below.
+
+~> **NOTE:** To to disable caching, do not provide the `cache` block in the configuration file.
+
+* `enabled` - (Optional) Is this Frontdoor Route enabled? Possible values are `true` or `false`. Defaults to `true`.
+
+* `forwarding_protocol` - (Required) The Protocol that will be use when forwarding traffic to backends. Possible values are `HttpOnly`, `HttpsOnly` or `MatchRequest`.
+
+* `https_redirect_enabled` - (Optional) Automatically redirect HTTP traffic to HTTPS traffic? Possible values are `true` or `false`. Defaults to `true`.
```

Comment:
should we add a note that this only work if `supported_protocols` has both set, HTTP and HTTPS. Tooke me a while to realize this :D not super intuitive

Reply:
@sebader, good point... I note would be helpful and I should also add a validation rule that verifies that these setting are set correctly and raise an error if the resource will not be doing what you expect it to be doing, because I agree it is not very intuitive to create the route resource correctly.

Reply:
yep sounds good